pkg/driver/aws-ebs/aws_ebs_tags_controller.go

```diff
@@ -0,0 +1,441 @@
+	awsSecretAccessKey, secretKeyFound := secret.Data["aws_secret_access_key"]
+
+	if accessKeyFound && secretKeyFound {
+		return createEC2ClientWithStaticKeys(awsRegion, string(awsAccessKeyID), string(awsSecretAccessKey))
+	}
+
+	// Otherwise, check for credentials field and create session using that
+	credentialsData, credentialsFound := secret.Data["credentials"]
+	if credentialsFound {
+		tempFile, err := writeCredentialsToTempFile(credentialsData)
+		if err != nil {
+			return nil, fmt.Errorf("error writing credentials to temporary file: %v", err)
+		}
+
+		return createEC2ClientWithCredentialsFile(awsRegion, tempFile)
+	}
+
+	return nil, fmt.Errorf("no valid AWS credentials found in secret")
+}
+
+// createEC2ClientWithStaticKeys creates an EC2 client using static credentials (access key and secret key)
+func createEC2ClientWithStaticKeys(awsRegion, awsAccessKeyID, awsSecretAccessKey string) (*ec2.EC2, error) {
+	awsSession, err := session.NewSession(&aws.Config{
+		Region:      aws.String(awsRegion),
+		Credentials: credentials.NewStaticCredentials(awsAccessKeyID, awsSecretAccessKey, ""),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error creating AWS session with static credentials: %v", err)
+	}
+	return ec2.New(awsSession), nil
+}
+
+// createEC2ClientWithCredentialsFile creates an EC2 client using a temporary credentials file
+func createEC2ClientWithCredentialsFile(awsRegion, credentialsFilename string) (*ec2.EC2, error) {
+	klog.Infof("Creating AWS session using credentials file: %s", credentialsFilename)
+
+	defer func() {
+		err := os.Remove(credentialsFilename)
+		if err != nil {
+			klog.Warningf("Failed to remove temporary credentials file: %v", err)
+		} else {
+			klog.Infof("Temporary credentials file %s removed successfully.", credentialsFilename)
+		}
+	}()
+
+	awsOptions := session.Options{
+		Config: aws.Config{
+			Region: aws.String(awsRegion),
+		},
+		SharedConfigState: session.SharedConfigEnable,
+		SharedConfigFiles: []string{credentialsFilename},
+	}
+
+	awsSession, err := session.NewSessionWithOptions(awsOptions)
+	if err != nil {
+		return nil, fmt.Errorf("error creating AWS session using credentials file: %v", err)
+	}
+
+	return ec2.New(awsSession), nil
+}
+
+// writeCredentialsToTempFile writes credentials data to a temporary file and returns the filename
+func writeCredentialsToTempFile(data []byte) (string, error) {
+	f, err := os.CreateTemp("", "aws-shared-credentials")
+	if err != nil {
+		return "", fmt.Errorf("failed to create file for shared credentials: %v", err)
+	}
+	defer f.Close()
+
+	if _, err := f.Write(data); err != nil {
+		defer os.Remove(f.Name())
+		return "", fmt.Errorf("failed to write credentials to %s: %v", f.Name(), err)
+	}
+	return f.Name(), nil
+}
+
+// getInfrastructure retrieves the Infrastructure resource in OpenShift
+func (c *EBSVolumeTagsController) getInfrastructure(ctx context.Context) (*configv1.Infrastructure, error) {
```

Comment:
why are we reading an object from informer with exponential backoff? See how we do this in other operators -  https://github.com/openshift/vmware-vsphere-csi-driver-operator/blob/master/pkg/operator/vspherecontroller/vspherecontroller.go#L167